### PR TITLE
Fix an reorderable list animation issue:"Reversed ReorderableListView drop animation moves item one row higher than it should #110949"

### PR DESCRIPTION
**Pull request by hannah-hyj**
fixed:#110949

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by hangyujin** [comment]
bump for review

File: packages/flutter/lib/src/widgets/reorderable_list.dart [hunk 1]
```diff
@@ -758,9 +758,12 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
 
   void _dragEnd(_DragInfo item) {
     setState(() {
-      if (_insertIndex! < widget.itemCount - 1) {
-        // Find the location of the item we want to insert before
+      if(_insertIndex == item.index){
```
Comment (darrenaustin) [inline packages/flutter/lib/src/widgets/reorderable_list.dart:761]: Formatting nit:

```suggestion
      if (_insertIndex == item.index) {
```

File: packages/flutter/lib/src/widgets/reorderable_list.dart [hunk 2]
```diff
@@ -758,9 +758,12 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
...
         _finalDropPosition = _itemOffsetAt(_insertIndex! + (_reverse ? 1 : 0));
+      }
+      else if (_insertIndex! < widget.itemCount - 1) {
```
Comment (darrenaustin) [inline packages/flutter/lib/src/widgets/reorderable_list.dart:764]: Formatting nit:
```suggestion
      } else if (_insertIndex! < widget.itemCount - 1) {
```

**Review by darrenaustin — APPROVED** [review]
LGTM. Nice fix.

Just a couple of formatting nits to fix. Thx.

**Pull request merged**